main.go

```diff
@@ -155,7 +160,32 @@ func main() {
 			ServerName:   *serverName,
 		})
 	}
-	db, err := sql.Open("mysql", *dsn)
+
+	var finalDsn = *dsn
+	if *prompt {
+		dsnRegex := regexp.MustCompile(`(\w*):?\w*(@.+)`)
```

Comment:
Can you add a comment explaining the regular expression? Experience taught me the hard way that even simple regular expression are easy to get wrong or misunderstand. :P